sphinx/contributing.rst

```diff
@@ -0,0 +1,139 @@
+
+Backend
+^^^^^^^
+
+Install backend dependencies with:
+
+.. code-block:: bash
+    
+    make install-skore
+
+
+You can run the API server with:
+
+.. code-block:: bash
+
+    make serve-api
+
+skore-ui
+^^^^^^^^
+
+Install skore-ui dependencies with:
+
+.. code-block:: bash
+    
+    npm install
+
+in the ``skore-ui`` directory.
+
+Run the skore-ui in dev mode (for hot-reloading) with
+
+.. code-block:: bash
+
+    npm run dev
+
+in the ``skore-ui`` directory
+
+Then, to use the skore-ui
+
+.. code-block:: bash
+
+    make build-skore-ui
+    make serve-ui
+
+PR format
+^^^^^^^^^
+
+We use the `conventional commits <https://www.conventionalcommits.org/en/v1.0.0/#summary>`_ format, and we automatically check that the PR title fits this format.
+In particular, commits are "sentence case", meaning "fix: Fix issue" passes, while "fix: fix issue" doesn't.
+
+Generally the description of a commit should start with a verb in the imperative voice, so that it would properly complete the sentence: "When applied, this commit will [...]".
+
+Example of correct commit: ``fix(docs): Add a contributor guide``.
+
+Contributing to the docstrings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When writing documentation, whether it be online, docstrings or help messages in the CLI and in the UI, we strive to follow some conventions that are listed below. These might be updated as time goes on.
+
+#. The docstring will be compiled using Sphinx numpydoc so use `RST (ReStructured Text) <https://docs.open-mpi.org/en/v5.0.x/developers/rst-for-markdown-expats.html>`_ for bold, URLs, etc.
+#. Argument descriptions should be written so that the following sentence makes sense: `Argument <argument> designates <argument description>`
+#. Argument descriptions start with lower case, and do not end with a period or other punctuation
+#. Argument descriptions start with "the" where relevant, and "whether" for booleans
+#. Text is written in US english ("visualize" rather than "visualise")
+#. In the CLI, positional arguments are written in snake case (``snake_case``), keyword arguments in kebab case (``kebab-case``)
+#. When there is a default argument, it should be shown in the help message, typically with ``(default: <default value>)`` at the end of the message
+
+Documentation
+-------------
+
+Our documentation uses `PyData Sphinx Theme <https://pydata-sphinx-theme.readthedocs.io/>`_.
+
+.. warning::
+
+    Modifications are to be done in the ``sphinx`` folder. The ``docs`` folder must *not* be touched!
+
+To build the docs:
+
+.. code-block:: bash
+
+    cd sphinx
+    make html
+
+Then, you can access the local build via:
+
+.. code-block:: bash
+
+    open build/html/index.html
+
+Help for common issues
+----------------------
+
+``make build-skore-ui`` doesn't work!
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Please check that your version of node is at least 20 using the following command:
+
+.. code-block:: bash
+
+    node -v
```

Comment:
```suggestion

```

This paragraph is weird in contributor guide, isn't it?

Reply:
It is kind of like a FAQ no? @augustebaum were the one who wrote this?

Reply:
I'm the one who wrote this! But I notice now that I didn't choose a good place...